### PR TITLE
[ENH] golden output tests for `HFTransformersForecaster`

### PR DESCRIPTION
**Pull request by geetu040**
Towards #10672

Add regression tests that validate huggingface transformers' sktime implementation outputs against outputs from the original/source implementation [autoformer](https://huggingface.co/docs/transformers/en/model_doc/autoformer), [informer](https://huggingface.co/docs/transformers/en/model_doc/informer), [time_series_transformer](https://huggingface.co/docs/transformers/en/model_doc/time_series_transformer).

<details>

<summary>
code used to generate the golden outputs for each case
</summary>

```py
"""Generate reference outputs directly from the transformers source implementation.

This script mirrors, from first principles based on the `transformers` docs
(`AutoformerForPrediction`/`InformerForPrediction`/`TimeSeriesTransformerForPrediction`
`forward`/`generate` signatures), what a user would do to run one of these pretrained
time-series-transformer checkpoints zero-shot on a series with exogenous time
features -- without importing
`sktime.forecasting.hf_transformers.HFTransformersForecaster`.

No sktime imports at all: the input series is a plain linear ramp, generated with
plain numpy, so this script has no dependency on sktime's datasets either.

Checkpoint choice matters here. HFTransformersForecaster always zeroes out
static-categorical and dynamic-real features (it doesn't support either), which for
a checkpoint pretrained *with* those (e.g. the `huggingface/*-tourism-monthly`
checkpoints) forces a real weight-shape mismatch in the value-projection layers,
which then has to be reinitialized and fine-tuned. To get a genuine zero-shot
comparison -- pretrained weights only, nothing randomly initialized or trained --
this script instead uses the `kashif/{informer,autoformer,time-series-transformer}
-traffic-hourly` checkpoints, which were pretrained with `num_static_categorical_
features=0` to begin with. Supplying exogenous features (`X`) with exactly
`num_time_features` columns then reproduces the checkpoint's original architecture
exactly, with zero shape mismatches.

Because there is no training and no reinitialized weight to seed, the only
randomness left is `.generate()`'s sampling from the predictive distribution, which
is seeded right before the call (mirroring HFTransformersForecaster's
`deterministic=True` behavior, which seeds internally at the same point).
"""

import numpy as np
import torch
import transformers
from torch import from_numpy

SAMPLE_SEED = 42  # seed applied right before sampling from the model


def build_series(n, num_time_features):
    """A deterministic linear ramp target with deterministic exogenous features."""
    y = np.arange(1, n + 1, dtype="float32")
    t = np.arange(n)
    x = np.stack(
        [((t + i) % (7 + i)) / (7 + i) for i in range(num_time_features)],
        axis=-1,
    ).astype("float32")
    return y, x


def build_config(model_path):
    """Build the no-static-feature config used by HFTransformersForecaster."""
    from transformers import AutoConfig

    config = AutoConfig.from_pretrained(model_path)
    config_dict = config.to_dict()
    config_dict["num_dynamic_real_features"] = 0
    config_dict["num_static_real_features"] = 0
    config_dict["num_static_categorical_features"] = 0
    config_dict["num_time_features"] = config.num_time_features
    if hasattr(config, "feature_size"):
        del config_dict["feature_size"]
    return config.from_dict(config_dict)


def run_reference_case(model_class, model_path, n=1000):
    """Zero-shot forecast with `model_class`, mirroring HFTransformersForecaster."""
    config = build_config(model_path)
    model, info = model_class.from_pretrained(
        model_path,
        config=config,
        output_loading_info=True,
        ignore_mismatched_sizes=True,
    )
    assert not info["mismatched_keys"], (
        f"expected a clean zero-shot load for {model_path}, "
        f"got mismatched keys: {info['mismatched_keys']}"
    )

    seq_len = config.context_length + max(config.lags_sequence)
    fh = config.prediction_length
    y, x = build_series(n, config.num_time_features)

    model.eval()
    hist = y.reshape((1, -1))
    hist_x = x[np.newaxis, -seq_len:, :]
    future_t = np.arange(n, n + fh)
    future_x = np.stack(
        [((future_t + i) % (7 + i)) / (7 + i) for i in range(config.num_time_features)],
        axis=-1,
    ).astype("float32")[np.newaxis, :, :]

    transformers.set_seed(SAMPLE_SEED)
    pred = model.generate(
        past_values=from_numpy(hist).to(model.dtype).to(model.device),
        past_time_features=from_numpy(hist_x).to(model.dtype).to(model.device),
        future_time_features=from_numpy(future_x).to(model.dtype).to(model.device),
        past_observed_mask=from_numpy((~np.isnan(hist)).astype(int)).to(model.device),
    )
    point_forecast = pred.sequences.mean(dim=1).detach().cpu().numpy().reshape(-1)
    # `model.generate(...)` returns predictions for true horizons 1..prediction_length
    # directly (no "cutoff"/horizon-0 slot), so these line up 1:1 with
    # `forecaster.predict(fh=[1, 2, 3, ...])`.
    return point_forecast[:3]


if __name__ == "__main__":
    from transformers import (
        AutoformerForPrediction,
        InformerForPrediction,
        TimeSeriesTransformerForPrediction,
    )

    cases = [
        (InformerForPrediction, "kashif/informer-traffic-hourly"),
        (AutoformerForPrediction, "kashif/autoformer-traffic-hourly"),
        (
            TimeSeriesTransformerForPrediction,
            "kashif/time-series-transformer-traffic-hourly",
        ),
    ]
    for model_class, model_path in cases:
        head = run_reference_case(model_class, model_path)
        print(model_path, head)

```

</details>

### CODE REVIEW

File: sktime/forecasting/hf_transformers.py [hunk 1]
```diff
@@ -249,7 +249,7 @@ def _fit(self, y, X, fh):
 
             if fh is not None:
                 _config["prediction_length"] = max(
-                    *(fh.to_relative(self._cutoff)._values + 1),
+                    *fh.to_relative(self._cutoff)._values,
```
Comment (geetu040) [inline sktime/forecasting/hf_transformers.py:252]: `fh.to_relative(...)` is already the 1-indexed count of steps ahead (e.g. `[1, 2, ..., 24]` for a 24-step horizon), so its max is exactly the required prediction_length, no "+1" needed.

File: sktime/forecasting/hf_transformers.py [hunk 2]
```diff
@@ -327,7 +327,7 @@ def _fit(self, y, X, fh):
 
         # Handle fine-tuning strategy
         if self.fit_strategy == "minimal":
-            if self.model is None and len(self.info["mismatched_keys"]) == 0:
+            if not any(param.requires_grad for param in self.model.parameters()):
                 return
```
Comment (geetu040) [inline sktime/forecasting/hf_transformers.py:331]: nothing left to adapt: every parameter was loaded from the checkpoint and frozen above, because its config matched the checkpoint's exactly and no layer had to be reinitialized. Training here would fail, as there is no gradient to take.

File: sktime/forecasting/hf_transformers.py [hunk 3]
```diff
@@ -421,7 +421,7 @@ def _predict(self, fh, X=None):
 
         pred = pd.Series(
             pred.reshape((-1,)),
-            index=ForecastingHorizon(range(len(pred)))
+            index=ForecastingHorizon(range(1, len(pred) + 1))
```
Comment (geetu040) [inline sktime/forecasting/hf_transformers.py:424]: `model.generate(...)` returns predictions for true horizons 1..len(pred) (genuine future steps beyond the cutoff, not including a "now" observation), so the horizons must be 1-indexed here to line up with what was actually predicted.

**Review by geetu040 — COMMENTED** [review]
leaving out some comments on the changes.

**Review by fkiraly — APPROVED** [review]
Good we did this, I suppose?
